yext/README.md

```diff
@@ -4,28 +4,108 @@ Connects Cohere to Yext Universal Search.
+
+To register the connector with Cohere's API using Oauth 2.0, use the following configuration.
+Also, you need to set the Authorization callback URL to `https://api.cohere.com/v1/connectors/oauth/token` on the App settings page.
+```bash
+ curl  -X POST \
+   'https://api.cohere.ai/v1/connectors' \
+   --header 'Accept: */*' \
+   --header 'Authorization: Bearer {COHERE-API-KEY}' \
+   --header 'Content-Type: application/json' \
+   --data-raw '{
+   "name": "Yext with OAuth",
+   "url": "{YOUR_CONNECTOR-URL}",
+   "oauth": {
+     "client_id": "{Your Yext App CLIENT-ID}",
+     "client_secret": "{Your Yext App SECRET}",
+     "authorize_url": "https://www.yext.com/oauth2/authorize",
+     "token_url": "https://api.yext.com/oauth2/accesstoken",
+   }
+ }'
+```
+
 
 The required environment variables for this connector are:
+```
+YEXT_AUTH_TYPE
+```
+This variable should be set to either `api_key` or `oauth` to use API Key or OAuth 2.0 authentication, respectively.
+
+```
+YEXT_API_URL
+```
+This variable should be set to the base URL of the Yext API. This is typically `https://cdn.yextapis.com`.
+
+
+```
+YEXT_API_KEY
+```
+This variable should be set to the API key of the app you created in Yext.
+Please note that this variable is needed only if you use API Key authentication.
+
+```
+YEXT_ACCOUNT_ID
+```
+This variable should be set to your Yext account ID.
+
+```
+YEXT_LOCALE
+```
+This variable should be set to the locale of the search experience you want to use. This is typically `en`.
+
+```
+YEXT_V
+```
+This variable should be set to the search version date in YYYYMMDD format. You can find this in the Yext UI under Search Experience >Test search.
+
 
-- `YEXT_API_KEY`
-- `YEXT_ACCOUNT_ID`
-- `YEXT_LOCAL`
-- `YEXT_EXPERIENCE_KEY`
-- `YEXT_V`
-- `YEXT_CONNECTOR_API_KEY`
 
-Optional environment variables are:
+See the [Yext documentation](https://hitchhikers.yext.com/docs/contentdeliveryapis/search/universalsearch/#operation/query) for more details.
 
-- `YEXT_LIMIT`
-- `YEXT_SOURCE`
-- `YEXT_VERSION`
+### Optional configuration
 
-Additionally, there are several optional environment variables. These variables, both required and optional,
-directly correspond to the parameters sent to Yext in the search request. They are documented in the
-Yext Universal Search: Query API documentation:
+```
+YEXT_VERSION
+```
+This variable should be set to the version of the Yext API you want to use. 
+The available versions are `STAGING` and `PRODUCTION` and the value should be set to `STAGING` for testing and `PRODUCTION` for production.
+The default value is `PRODUCTION`.
+
+```
+YEXT_SEARCH_LIMIT
+```
+This variable may contain a JSON object specifying the limit for each vertical. 
+Each key is a vertical key and the value for each of those keys is a number 1-50 that denotes the limit for that vertical. 
+If a vertical is not specified, the default limit is 10 for all verticals
+
+```
+YEXT_RESTRICT_VERTICALS
```

Comment:
Both the `SEARCH_LIMIT` and `RESTRICT_VERTICALS` variables mention verticals, I would either document the full list of verticals available by Yext here for clarity or link the documentation, to me it seems unclear what possible options I can set

Reply:
Thanks